## Text Classification

Sentiment Analysis, Text classification with positive or negative cases.

In [3]:
import nltk
import random
from nltk.corpus import movie_reviews

documents list is for creating training and testing sets.

In [4]:
documents = [(list(movie_reviews.words(fileid)), category)
                for category in movie_reviews.categories()
                for fileid in movie_reviews.fileids(category)]

random.shuffle(documents)

print(documents[1])

(['it', 'has', 'happened', 'again', '.', 'a', 'movie', 'that', 'is', 'well', 'scripted', ',', 'well', 'acted', ',', 'and', 'well', 'directed', '.', 'it', 'is', 'involving', ',', 'suspenseful', '.', '.', '.', 'it', 'even', 'has', 'an', 'unexpected', 'twist', 'at', 'the', 'end', 'that', 'packs', 'a', 'punch', 'like', 'a', 'blow', 'to', 'the', 'solar', 'plexus', '.', 'but', 'in', 'order', 'to', 'accept', 'its', 'premise', 'one', 'must', 'also', 'accept', 'a', 'spiritual', 'lie', '.', 'the', 'sixth', 'sense', 'is', 'that', 'movie', '.', 'starring', 'bruce', 'willis', '(', 'armageddon', ')', 'as', 'a', 'child', 'psychologist', 'and', 'haley', 'joel', 'osment', '(', 'forrest', 'gump', ')', 'as', 'a', 'troubled', 'youngster', ',', 'the', 'sixth', 'sense', 'is', 'more', 'spooky', 'than', 'scary', ',', 'though', 'it', 'does', 'contain', 'disturbing', 'images', 'designed', 'to', 'startle', '.', 'eight', '-', 'year', '-', 'old', 'cole', 'sear', '(', 'osment', ')', 'sees', 'dead', 'people', '.', '

_(Above content can differ as it is randomized.)_

### Frequency Distribution of words

In [7]:
all_words = []
for w in movie_reviews.words():
    all_words.append(w.lower())

all_words = nltk.FreqDist(all_words)
print(all_words.most_common(15))

[(',', 77717), ('the', 76529), ('.', 65876), ('a', 38106), ('and', 35576), ('of', 34123), ('to', 31937), ("'", 30585), ('is', 25195), ('in', 21822), ('s', 18513), ('"', 17612), ('it', 16107), ('that', 15924), ('-', 15595)]


In [8]:
print(all_words["stupid"]) 

253


_(The word stupid occurs 253 times in the movie reviews.)_